includes/data-stores/class-wc-admin-reports-products-data-store.php

```diff
@@ -337,36 +337,103 @@ public static function sync_order_products( $order_id ) {
 			$order_item_id     = $order_item->get_id();
 			$quantity_refunded = isset( $refunds[ $order_item_id ] ) ? $refunds[ $order_item_id ]['quantity'] : 0;
 			$amount_refunded   = isset( $refunds[ $order_item_id ] ) ? $refunds[ $order_item_id ]['subtotal'] : 0;
+			$product_qty       = $order_item->get_quantity( 'edit' ) - $quantity_refunded;
+
+			// Shipping amount based on woocommerce code in includes/admin/meta-boxes/views/html-order-item(s).php
+			// distributed simply based on number of line items.
```

Comment:
Would it be better for this logic to live inside another util/class so it could be reused elsewhere - or allow for some test coverage?

Reply:
I think that should be great a new class or function, so we can use in both places, also we can include unit tests.

Reply:
Factored the logic out to a class extending `WC_Order` in https://github.com/woocommerce/wc-admin/pull/1286/commits/f1dddc9243db0cc6808d9a13ecd4beb350d40ae4. 
Would be great to later add unit tests for it if time allows.